pages/en/zkapps/how-to-write-a-zkapp.mdx

```diff
@@ -207,121 +205,192 @@ For a full list, see the <DocLink copy="SnarkyJS reference" url="/zkapps/snarkyj
 
 #### Smart Contract
 
-Let’s put this all together and create a basic smart contract. This where you define your custom logic.
+<!-- TODOs
+  IMO, needs to be fleshed out a bit more before sending users to "test your zkApp" or "advanced topics".
+  Basics that are missing:
+  * payments
+  * Mina.transaction / at least hinting at the basic mechanics of how to compile / prove
+  * permissions? (maybe an advanced topic)
+ -->
+
+Now that we have covered the basics of writing SnarkyJS programs, let's see how to create a smart contract.
 
 Smart contracts are written by extending the base class `SmartContract`:
 
 ```ts
 class HelloWorld extends SmartContract {}
 ```
 
-A smart contract can contain **on-chain state**, which is declared as a property
-on the class with the `@state` decorator:
-
-```ts
-class HelloWorld extends SmartContract {
-  @state(Field) x = State<Field>();
-}
-```
-
-Here, `x` is of type `Field`. Only types built out of Field can be used for state variables.
-In the current design, the state can consist of at most 8 Fields of 32 bytes each.
-States are initialized with the `State()` function.
-
 The `constructor` of a `SmartContract` is inherited from the base class and should not be overriden.
 It takes the zkApp account address (a public key) as its only argument:
 
 ```ts
 let zkAppKey = PrivateKey.random();
 let zkAppAddress = PublicKey.fromPrivateKey(zkAppKey);
 
-let zkApp = new SmartContract(zkAppAddress);
+let zkApp = new HelloWorld(zkAppAddress);
 ```
 
-##### SnarkyJS Deployment Method
+Later, we show you how you can _deploy_ a smart contract to an on-chain account.
+
+<Alert kind="note">
+  On Mina, there is no strong distinction between normal "user accounts" and
+  "zkApp accounts". A zkApp account is just a normal account that has a smart
+  contract deployed to it -- which essentially just means there's a verification
+  key stored on the account, which can verify off-chain execution proofs.
+</Alert>
+
+##### Methods
 
-There is one method that every smart contract needs: `deploy`. It describes how the zkApp account
-will be initialized when the zkApp is deployed:
+Interaction with a smart contract happens by calling on or more of its _methods_. You declare methods using the `@method` decorator:
 
 ```ts
 class HelloWorld extends SmartContract {
-  @state(Field) x = State<Field>();
-
-  deploy(initialBalance: UInt64, x: Field) {
-    super.deploy();
-    this.balance.addInPlace(initialBalance);
-    this.x.set(x);
+  @method myMethod(x: Field) {
+    x.mul(2).assertEquals(5);
   }
 }
 ```
 
-The deploy method needs one default parameter, the initial balance of the zkApp account.
-The first two lines below are mandatory:
+In a method, you can use SnarkyJS commands like `Field.mul` and `Field.assertEquals` to define your custom logic.
+
+Later, we'll show how you can...
+
+- run a method off-chain
```

Comment:
`run a method off-chain` > `run a method (off-chain)`

Small suggestion for clarity. I think the latter better indicates that it always runs off chain, rather than us just teaching this aspect as if on-chain execution were also possible